src/components/text-editor/utils/markdown-converter.ts

```diff
@@ -1,8 +1,32 @@
 import { ContentTypeConverter } from './content-type-converter';
 import { EditorView } from 'prosemirror-view';
-import { defaultMarkdownSerializer } from 'prosemirror-markdown';
+import {
+    MarkdownSerializer,
+    defaultMarkdownSerializer,
+} from 'prosemirror-markdown';
 import { markdownToHTML } from '../../markdown/markdown-parser';
 
+const customMarkdownSerializer = new MarkdownSerializer(
+    {
+        ...defaultMarkdownSerializer.nodes,
+    },
+    {
+        ...defaultMarkdownSerializer.marks,
+        underline: {
+            open: '<u>',
+            close: '</u>',
+            mixable: true,
+            expelEnclosingWhitespace: true,
+        },
+        strikethrough: {
+            open: '~~',
+            close: '~~',
+            mixable: true,
+            expelEnclosingWhitespace: true,
```

Comment:
Have we decided anything on the markdown vs html output question?

I'm asking since we seem to be mixing html (`<u>undeline</u>`) and markdown (`~~strikethrough~~`) here…? I know vanilla markdown doesn't actually support underline, while some flavours support interpreting enclosing single underscores as underline (which I've always thought makes sense 🤷‍♂️).

Reply:
Oh… `customMarkdownSerializer`… so I guess this is when outputting markdown then?

Reply:
This is when `markdown` is set as the `contentType`, but your question does have me thinking further about this.

Reply:
Are we outputting markdown or html by default?

I would assume the main focus right now is html anyway, since that's presumably what the email feature needs?

Reply:
We are outputting markdown by default but that can be changed by modifying one line in the `prosemirror-adapter`

Reply:
Should we hold off on including the underline button?

I mean, it will work if the output is formatted with `limel-markdown`, as that can handle mixed markdown and html. But we're not actually outputting pure markdown when there's something that's been underlined. So I'm thinking we might not want to include the underline button in the default set of menu buttons when the output mode is markdown…

But we don't want to spend any time on including it when in html mode and excluding it in markdown mode. Not right now.

Reply:
I'm thinking I can just move that commit to a separate branch, and you can open a PR for that commit only, and then we can leave that to rest until we're ready to look at it.

Reply:
I think that sounds like a good solution.